desc/optimize/_constraint_wrappers.py

```diff
@@ -292,8 +283,14 @@ def jac_unscaled(self, x_reduced, constants):
             Jacobian matrix.
         """
         x = self.recover(x_reduced)
-        df = self._objective.jac_unscaled(x, constants)
-        return df[:, self._unfixed_idx] @ self._Z
+        if self._objective._deriv_mode == "blocked":
```

Comment:
What is deriv_mode == "blocked"? How is it dfferent from fwd and rev?

Reply:
https://github.com/PlasmaControl/DESC/blob/d4ab2f58ff16c388a432a859016ba1c4122ed31e/desc/objectives/objective_funs.py#L34

Reply:
though with this change is batched still the most memory intensive? as now we can do the jvp's when batched is the method, while blocked is still making the whole jacobian, so it strikes me as that might be more intensive

Reply:
blocked still uses JVPs, but does them on the individual objectives rather than on the whole. The only exception is objectives that specify reverse mode, in which case we calculate that full jacobian. This was done because some objectives may _only_ support reverse mode (like ODE integration). But in those cases its usually a scalar or very low dimensional objective so doing the full reverse mode jacobian is pretty cheap (and possibly cheaper to doing multiple jvps, depending on how many variables we're optimizing)